### DISCUSSION TITLE
Koreader support on newer kindle firmwares with hard floats?

### DISCUSSION THREAD

**Discussion by arooni** [koreader/koreader#11298] [General]
Asked this question on mobileread:
>Is there typically ever an ability to upgrade firmware without losing the jailbreak? I prefer having koreader and the jailbreak compared to the newer scribe notebook features in 5.16.5 so perhaps it's a moot point. But should I make the assumption that upgrading the firmware means losing the jailbreak ? If so, I should kiss the idea of an upgrade goodbye and stop reading about the new firmware updates lol.

I think the notebook features are good enough for me currently to use for the foreseeable future but who doesn't like shiny new things ? Lol

And got this answer;

> I repeat: upgrading past 5.16.2 is NOT the end of the JB, it can be restored. HOWEVER, nothing works properly on these firmwares, because Amazon switched their compiling backend mechanism (from softfloats to hardfloats), meaning that currently compiled code in KOReader and all packages won’t work with 99% chance. Unless developers are willing to widely support this new architecture (which I doubt), you can’t have everything at once.

I assume there is never going to be support for this hardfloat architecture in later firmwares on Kindle?

**Comment by NiLuJe** [comment]
Not for a while, no.

**Reply by Frenzie** [reply]
Though I'll briefly point out there are various armhf binaries that aren't guaranteed not to work xp

**Reply by NiLuJe** [reply]
More to the point is that a JB is useless on those FW versions, because nothing is built for it, so *nothing* works. KOReader would be the *least* of your problems ;).

**Reply by Frenzie** [reply]
Well yes, that would be a bit of an obstacle. :rofl:

**Reply by Commodore64user** [reply]
are we there yet?

**Comment by mergen3107** [comment]
> because nothing is built for it

@NiLuJe 
Do you mean by this, for example, USBnetwork and its whole bin folder? :D 

Any hints where I can start looking for recompile?

**Comment by shamanNS** [comment]
@mergen3107 
> Any hints where I can start looking for recompile?

Maaaybe look here:

https://github.com/notmarek/koxtoolchain/commit/0f47e4b134ca8828d994544f3473e9cfbf18fde7
https://github.com/notmarek/crosstool-ng/tree/58e622ba27a43c119eaf3ac3ddf2f703bc7ba465

@notmarek

**Reply by mergen3107** [reply]
Interesting! Thank you!

**Reply by notmarek** [reply]
just so you are aware i wasnt able to compile koreader with the toolchain i have however had success with fbink and gandalf :)

**Reply by mergen3107** [reply]
@notmarek 
KOReader-hf package from your hf website shows 403 error (forbidden) 
https://fw.notmarek.com/khf/

**Comment by notmarek** [comment]
Ah yeah sorry about that sed script apparently broke - i will fix it in a
few minutes as well as build latest nightly thanks for letting me know

**Reply by mergen3107** [reply]
No worries! I just noticed it :D

**Reply by notmarek** [reply]
sooooo, turns out it wont be as smooth sailing, half of my build vm got wiped during an unfortunate power off event

**Reply by Frenzie** [reply]
Ouch, my condolences.

**Reply by notmarek** [reply]
A new build vm is up and running and https://fw.notmarek.com/khf/koreader redirects to the latest build

**Reply by mergen3107** [reply]
Thank you!

**Comment by shamanNS** [comment]
@Frenzie 
@benoit-pierre 
@NiLuJe 

Can someone explain why official KOReader releases still aren't building files for "kindlehf" target? More than a month after relevant changes have been merged. 

[Frenzie's comment](https://github.com/koreader/koreader/pull/11986#issuecomment-2165765057)

What are the remaining problems and what is not " a proper toolchain" in the merged changes? And is anyone planning to do the necessary work to the build systems you're using to build KOReader releases?

**Reply by mergen3107** [reply]
AFAIR NiLuJe wanted to tweak toolchain a bit. 

I am happy to test any changes!

**Reply by NiLuJe** [reply]
Yep, going to work on updated toolchains sometime in the summer, so it's one less pointless update of the docker images to wait until that's done ;).

Plus, there shouldn't really any users on affected FW for now anyway.

**Reply by mergen3107** [reply]
Thanks to Marek, there might be some :D 
His koreader-hf package even directs to his server with hf mirrored nightlies :D (haven’t tested yet, but soon™)
https://fw.notmarek.com/khf/

(although I admit to be belong more to a beta tester rather than a “user” xD)

**Reply by shamanNS** [reply]
Summer of year 2024. hopefully? 😋
